### ISSUE TITLE
Screen Batching Compat Issue

### ISSUE THREAD

**Issue by Julienraptor01** [RaphiMC/ImmediatelyFast#138]
i have found an issue with the new optimisation
but i'm running 1.19.2 with a fork of a mod so it might not even have the latest way to achieve what it does
so i'm running this version of skyblocker https://github.com/Julienraptor01/Skyblocker/releases/tag/v1.8.10-beta with this version of IF https://github.com/RaphiMC/ImmediatelyFast/actions/runs/6776175155 on 1.19.2 and i have this issue
The 1st picture is the expected behavior with the opti disabled and the 2nd one is with it enabled
basically our slot highlighter is supposed to be transparent, but if the optimisation is enabled, it's opaque
![image](https://github.com/RaphiMC/ImmediatelyFast/assets/49841713/d9eec22f-c774-4486-9df9-926710c98ef9)
![image](https://github.com/RaphiMC/ImmediatelyFast/assets/49841713/f14a98e6-fdf5-4260-a4ad-d82237f3e81c)

**Comment by RaphiMC** [comment]
Not really fixable with the current batching system where the item overlay and the item itself are rendered seperately. Mods which want to modify the overlay currently have to explicitly render into the overlay buffer because its not easily possible to detect that in ImmediatelyFast. You would have to call BatchingBuffers.beginItemOverlayRendering(); before rendering the overlay and call BatchingBuffers.endItemOverlayRendering(); after you rendered the overlay. Here are the IF API docs: https://github.com/RaphiMC/ImmediatelyFast/blob/1.20/API_DOCS.md
